openmc/data/resonance.py

```diff
@@ -0,0 +1,1063 @@
+    energy_min : float
+        Minimum energy of the resolved resonance range in eV
+    energy_max : float
+        Maximum energy of the resolved resonance range in eV
+    channel : dict
+        Dictionary whose keys are l-values and values are channel radii as a
+        function of energy
+    scattering : dict
+        Dictionary whose keys are l-values and values are scattering radii as a
+        function of energy
+
+    Attributes
+    ----------
+    channel_radius : dict
+        Dictionary whose keys are l-values and values are channel radii as a
+        function of energy
+    energy_max : float
+        Maximum energy of the resolved resonance range in eV
+    energy_min : float
+        Minimum energy of the resolved resonance range in eV
+    scattering_radius : dict
+        Dictionary whose keys are l-values and values are scattering radii as a
+        function of energ
+    target_spin : float
+        Intrinsic spin, :math:`I`, of the target nuclide
+
+    """
+    def __init__(self, target_spin, energy_min, energy_max, channel, scattering):
+        self.target_spin = target_spin
+        self.energy_min = energy_min
+        self.energy_max = energy_max
+        self.channel_radius = channel
+        self.scattering_radius = scattering
+
+        self._prepared = False
+        self._parameter_matrix = {}
+
+    @classmethod
+    def from_endf(cls, ev, file_obj, items):
+        """Create resonance range from an ENDF evaluation.
+
+        This factory method is only used when LRU=0, indicating that only a
+        scattering radius appears in MF=2, MT=151. All subclasses of
+        ResonanceRange override this method with their own.
+
+        Parameters
+        ----------
+        ev : openmc.data.endf.Evaluation
+            ENDF evaluation
+        file_obj : file-like object
+            ENDF file positioned at the second record of a resonance range
+            subsection in MF=2, MT=151
+        items : list
+            Items from the CONT record at the start of the resonance range
+            subsection
+
+        Returns
+        -------
+        openmc.data.ResonanceRange
+            Resonance range data
+
+        """
+        energy_min, energy_max = items[0:2]
+
+        # For scattering radius-only, NRO must be zero
+        assert items[4] == 0
+
+        # Get energy-independent scattering radius
+        items = get_cont_record(file_obj)
+        target_spin = items[0]
+        ap = Polynomial((items[1],))
+
+        # Calculate channel radius from ENDF-102 equation D.14
+        a = Polynomial((0.123 * (1.008665*ev.target['mass'])**(1./3.) + 0.08,))
+
+        return cls(target_spin, energy_min, energy_max, {0: a}, {0: ap})
+
+    def reconstruct(self, energies):
+        """Evaluate cross section at specified energies.
+
+        Parameters
+        ----------
+        energies : float or Iterable of float
+            Energies at which the cross section should be evaluated
```

Comment:
Tangentially related to this (I should probably just put this in a separate issue), I've long had the idea of incorporating units in our Python API for input generation. For example, having something like:
```Python
from openmc.units import inch
import openmc

s = openmc.Sphere(R=3.0*inch)
```
Rather than expecting a real number, the `R` argument in this case would expect a physical quantity with units of length. On the backend, the API would automatically convert the user-specified units to canonical units (in this case, centimeters) when generating XML files. I think it would be pretty cool, but probably not a trivial amount of work to implement.